### PR TITLE
frontend create account pages + created root style

### PR DESCRIPTION
**Pull request by throwExceptionPlease**
added frontend aspects for the create account pages. also added CSS variables and root styles. all create-account pages are in src > create-acc-pages and the logo was added in an assets folder.

### CODE REVIEW

**Comment by ardada2468** [comment]
Hey,

The UI looks great but I think we may be changing the login page based on the meeting yesterday. But while the designer are working on the changes could you fix the following: 

- I see that you are checking if the password meets the requirements but I cannot see the checkbox UI password validation on the latest figma design. 
- The form does not do email validation, I was able to put in any random string and it was accepting it.

**Comment by throwExceptionPlease** [comment]
Added checking for both password fields to contain the same value; added check to prevent user from submitting form without satisfying the pwd reqs; changed input type from text to email for email inputs (oops); updated colors in root style as per figma; added checkbox ui password validation in both pages

not yet implemented: adding the icon and red border for invalid confirmation password, and new styling for when a pwd req has yet to be fulfilled

will work on some more later tonight

**Comment by throwExceptionPlease** [comment]
Finished with everything except one, which is getting the minus sign in the checkbox

**Comment by aduvvuri007** [comment]
I added all the backend code onto this branch to make it easier. I added the function to create accounts with email and password, added user data to the firestore database and their specified collection(applicants and reviewers) and then I created the cloud functions to add custom claims to users (functions not deployed yet and still not tested)

File: react-app/ccf/src/create-acc-pages/create-acc-reviewer/CreateAccReviewer.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        role: 'reviewer',
+      });
+      await addReviewerRole({email: email})
+      .then((result) => {
+        console.log(result.data);  // Success message from the function
+      })
+      .catch((error) => {
+        console.log('Error: ', error);
+      });
+      console.log(user);
```
Comment (ardada2468) [inline react-app/ccf/src/create-acc-pages/create-acc-reviewer/CreateAccReviewer.tsx:78]: We should not be printing this information in the console.

**Review by ardada2468 — CHANGES_REQUESTED** [review]
Great Work Guys, This is almost ready to be merged but just needs a couple of edits: 

1. We need to do email validation because I can still hit the sumbit button when i only put ardada2468 (error shows up in console).

image

2. Remove all instances where we are console.logging the user information from firebase. 

image

File: functions/src/index.ts [hunk 2]
```diff
@@ -17,3 +20,32 @@ import * as logger from "firebase-functions/logger";
...
 // });
+export const addReviewerRole = functions.https.onCall((data: any, context: any) => {
+    return admin.auth().getUserByEmail(data.email).then((user: any) => {
+        return admin.auth().setCustomUserClaims(user.uid, {
+            role: "reviewer"
+        });
+    }
+    ).then(() => {
+        return {
+            message: `Success! ${data.email} has been made an admin.`
```
Comment (jdlane) [inline functions/src/index.ts:31]: Small thing, this returns "has been made an admin" for any user role. It should say "reviewer" for reviewer etc.

**Comment by throwExceptionPlease** [comment]
Added email validation. There was no design (and technical) guide for indicating an invalid email, so I just reused what was done for the password.

**Review by ardada2468 — CHANGES_REQUESTED** [review]
This fixes all of the issues but if you could just add 2 more things: 

After account creation, automatically redirect the user to the sign-in page.

Disable the "Submit" button (gray it out) if any required field is not completed or does not pass validation. The button should remain unclickable until:

- All fields are filled out.
- All entries have passed validation checks.

Once all fields are validated, the button becomes clickable, enabling the user to submit the form.

**Review by jdlane — COMMENTED** [review]
We talked in the meeting about greying out the submit button until the fields are all valid. Could you add that change before we merge?

**Comment by throwExceptionPlease** [comment]
added extra conditionals to the submit button to ensure no field is left behind or invalid (this ones for pwds + email) before the user is able to click on the submit button. not quite sure if u needed me to validate the first name, last name, and affiliation too so i left them be. didn't see any new updates on the login stuff when i did the pull, so i rerouted the submit button from the dashboards to the login page as specified in app.tsx

let me know if i missed anything.

**Review by jdlane — CHANGES_REQUESTED** [review]
Nice job, just one last fix. We need the creating account logic in the CreateAcc files to be moved to separate functions in a new AccountManager file in the functions folder. That way we avoid repeated code and can easily find account functions to make any changes in the future. Also, instead of creating new functions and firestore objects, you should import the existing ones from index.ts

**Review by jdlane — COMMENTED** [review]
Also, to keep things consistent we should probably either make a "create-acc-applicant" folder or move the create reviewer files out of their own folder.

**Comment by throwExceptionPlease** [comment]
CreateAccApplicants and CreateAccReviewers are now in their separate folders in the create-acc-pages folder. I realized that createAccApplicants didn't have the pwd validation text that createAccReviewers had so I added it.

**Comment by ardada2468** [comment]
This is great, please fix the merge conflict and we are ready to merge.

**Comment by jdlane** [comment]
> This is great, please fix the merge conflict and we are ready to merge.

I'm going to go ahead and resolve those myself, just so we get that through. Nice job guys!

**Review by jdlane — APPROVED** [review]

**Pull request merged**